lib/collection/src/collection_manager/segments_searcher.rs

```diff
@@ -342,58 +342,90 @@ impl SegmentsSearcher {
     /// The points ids can contain duplicates, the records will be fetched only once
     ///
     /// If an id is not found in the segments, it won't be included in the output.
-    pub fn retrieve(
-        segments: &RwLock<SegmentHolder>,
+    pub async fn retrieve(
+        segments: LockedSegmentHolder,
+        points: &[PointIdType],
+        with_payload: &WithPayload,
+        with_vector: &WithVector,
+        runtime_handle: &Handle,
+    ) -> CollectionResult<HashMap<PointIdType, Record>> {
+        let stopping_guard = StoppingGuard::new();
+        runtime_handle
+            .spawn_blocking({
```

Comment:
For reviewers, important part: here we now spawn blocking tasks for each segment.

Reply:
Having one task per segment is possible optimization (marked by a TODO below)

For now there is one task per `segment_holder`, so just one per request :)